packages/tracker/tracker.js

```diff
@@ -196,6 +196,16 @@ Tracker.Computation = class Computation {
     this._onError = onError;
     this._recomputing = false;
 
+    /**
+     * @summary Forces autorun blocks to be executed in synchronous-looking order by storing the value autorun promise thus making it awaitable.
+     * @locus Client
+     * @memberOf Tracker.Computation
+     * @instance
+     * @name  firstRunPromise
+     * @returns {Promise<unknown>}
+     */
+    this.firstRunPromise = undefined;
```

Comment:
I would try thinking of a more suitable name for this property, `firstRunPromise` sounds good but what about `invocationPromise` or something like that, not sure if you guys have any ideas(cc: @radekmie @DanielDornhardt ), `firstRunPromise` does not seem to express exactly what this does.

Reply:
I honestly think it makes sense, but if you have a different one - let's use that. (It may not be needed, though - I'll add a comment below.)

Reply:
I like `firstRunPromise`, i think it pretty straightforward in what it does... and it also tells the story that this will only be for the first run, for example it expresses better that even if you use guard clauses and stuff this'll no longer help you...

```js
Tracker.autorun(async() {
    if (await Meteor.loggingInAsync()) {
        return // this is what the promise will resolve to
    } 
    if (!await Meteor.user().isAllowedHere()) {
        return // this will not get its own promise
    } 
    // lots of code
    // will happen WAY after the `firstRunPromise` / `invocationPromise` has been resolved
    await a...
    await b...
})
```